lib/CL/devices/ventus/pocl_ventus.cc

```diff
@@ -249,11 +249,14 @@ pocl_ventus_init (unsigned j, cl_device_id dev, const char* parameters)
   dev->linker_available = CL_TRUE;
 
   char extensions[1024];
+  char features[1024];
   extensions[0] = 0;
-  strcat (extensions, "cl_khr_fp64"
-                      " __opencl_c_generic_address_space");
-  dev->extensions = strdup (extensions); // no extention support now
-
+  features[0] = 0;
+  strcat (extensions, "cl_khr_fp64 ");
+  strcat(features, "__opencl_c_generic_address_space "
+                   " __opencl_c_named_address_space_builtins");
+  dev->extensions = strdup (extensions);
+  dev->features = strdup(features);
```

Comment:
Maybe you can try to delete this part of the code modification and keep only the 1156 lines of code to see if the effect is equivalent.